tmk_core/rules.mk

```diff
@@ -670,17 +670,34 @@ all-keyboards-defaults-%:
 all-keyboards-defaults: all-keyboards-defaults-all
 
 KEYBOARDS := $(SUBDIRS:$(TOP_DIR)/keyboard/%/=/keyboard/%)
-all-keyboards-%: $(KEYBOARDS)
-/keyboard/%:
-	$(eval KEYBOARD=$(patsubst /keyboard/%,%,$@))
-	$(eval KEYMAPS=$(notdir $(patsubst %/.,%,$(wildcard $(TOP_DIR)$@/keymaps/*/.))))
-	@for x in $(KEYMAPS) ; do \
-		printf "Compiling $(BOLD)$(KEYBOARD)$(NO_COLOR) with $(BOLD)$$x$(NO_COLOR)" | $(AWK) '{ printf "%-88s", $$0; }'; \
-		LOG=$$($(MAKE) -C $(TOP_DIR)$@ $(subst all-keyboards-,,$@) keymap=$$x VERBOSE=$(VERBOSE) COLOR=$(COLOR) SILENT=true 2>&1) ; if [ $$? -gt 0 ]; then $(PRINT_ERROR_PLAIN); elif [ "$$LOG" != "" ] ; then $(PRINT_WARNING_PLAIN); else $(PRINT_OK); fi; \
-	done
-
+all-keyboards-all: $(addsuffix -all,$(KEYBOARDS))
+all-keyboards-quick: $(addsuffix -quick,$(KEYBOARDS))
+all-keyboards-clean: $(addsuffix -clean,$(KEYBOARDS))
```

Comment:
@algernon 

That is what I thought too, and I tried many variations of this yesterday, and now again, this is what I get

```
$ make all-keyboards-quick
make: *** No rule to make target 'all-keyboards-quick'.  Stop.
```

If I do this instead though 
```
$ make all-keyboards-quick
make: *** No rule to make target 'all-keyboards-quick'.  Stop.
```
It prints out the correct rule names, so the prerequisites pattern is right. I also tried adding the debug flag to make, and it didn't help me debug the problem either. But maybe someone with better makefile programming skills could?